GameRunner/Program.cs

```diff
@@ -197,18 +209,18 @@ ScriptsOfTribute.AI.ScriptsOfTribute PrepareGame(AI bot1, AI bot2, LogsEnabled e
 
     if (noOfThreads == 1)
     {
-        Console.WriteLine($"Running {runs} games - {bot1Type!.Name} vs {bot2Type!.Name}");
+        Console.WriteLine($"Running {runs} games - {bot1Info!.BotType.Name} vs {bot2Info!.BotType.Name}");
 
         var counter = new GameEndStatsCounter();
 
         var timeMeasurements = new long[runs];
 
         var granularWatch = new Stopwatch();
         var currentSeed = actualSeed;
+        var bot1 = bot1Info.IsExternal ? (AI?)Activator.CreateInstance(bot1Info.BotType, bot1Info.ProgramName, bot1Info.FileName) : (AI?)Activator.CreateInstance(bot1Info.BotType);
```

Comment:
I moved instantianting bots here because in my opinion bots should be able to learn from their mistakes from previous games. By creating them for every game we block it, unless they start using some local files etc. 
The functions players have to implement are stateless (PickPatron, Play) so it doesn't hurt doing it, it may also increase usability of GameEnd function in AI class.